netflix/src/components/main/sort/GenreOption.tsx

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { setCurrentGenre, setFilms } from "../../../redux/films";
+import { useDispatch } from "react-redux";
+import { filterGenre } from "../../../redux/api";
+
+const GenreOption = (props: any) => {
+  const dispatch = useDispatch();
+  const genre = props.genre.toString();
+  const [trigger] = filterGenre(genre);
+
+  const sortGenre = async () => {
+    const payload = await trigger({ genre }).unwrap();
```

Comment:
You have bug here... you do not send `sortBy` value, so whenever you change `genre` filter your `sortBy` dropdown is not considered anymore.

Reply:
Also your filter value from TopBar is not considered on this request... so either reset its value or send with request.